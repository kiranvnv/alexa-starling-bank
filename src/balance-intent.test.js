const BalanceIntent = require('./balance-intent');

describe('handle', () => {
  describe('getting a balance from Starling successfully', () => {
    it('returns a valid Alexa response', () => {
      const client = {
        getBalance: jest.fn(() => Promise.resolve({ data: { amount: 1000 } })),
      };

      const handler = new BalanceIntent(client);

      const mockCallback = jest.fn().mockImplementation((_, response) => {
        expect(client.getBalance).toHaveBeenCalled();
        expect(response).toMatchSnapshot();
      });

      handler.handle({}, {}, mockCallback);
    });
  });

  describe('failing to get a balance from Starling', () => {
    it('returns a valid Alexa response', () => {
      const client = {
        getBalance: jest.fn(() => Promise.reject()),
      };

      const handler = new BalanceIntent(client);
      const mockCallback = jest.fn().mockImplementation((_, response) => {
        expect(client.getBalance).toHaveBeenCalled();
        expect(response).toMatchSnapshot();
      });

      handler.handle({}, {}, mockCallback);
    });
  });
});

