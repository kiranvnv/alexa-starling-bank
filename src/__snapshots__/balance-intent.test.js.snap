// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handle failing to get a balance from Starling returns a valid Alexa response 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "Something went wrong. Check your Starling access token, re-deploy, and then try again",
      "type": "PlainText",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;

exports[`handle getting a balance from Starling successfully returns a valid Alexa response 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "Your balance is Â£1000.",
      "type": "PlainText",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;
